syntax = "proto3";

package TW.THORChain.Proto;
option java_package = "wallet.core.jni.proto";

import "Cosmos.proto";

message Message {
    oneof message_oneof {
        Cosmos.Proto.Message.Send send_coins_message = 1;
    }
}

// Input data necessary to create a signed order.
message SigningInput {
    uint64 account_number = 1;
    string chain_id = 2;
    Cosmos.Proto.Fee fee = 3;
    string memo = 4;
    uint64 sequence = 5; 

    bytes private_key = 6;

    repeated Message messages = 7;

    Cosmos.Proto.BroadcastMode mode = 8;
}

// Transaction signing output.
message SigningOutput {
    // Signature
    bytes signature = 1;
    // Signed transaction in JSON.
    string json = 2;
}
